tests/vlan/test_vlan.py

```diff
@@ -195,21 +195,24 @@ def tearDown(vlan_ports_list, duthost, ptfhost, vlan_intfs_list, portchannel_int
     ptfhost.command('supervisorctl stop arp_responder')
 
     logger.info("Delete VLAN intf")
-    try:
-        for item in vlan_ports_list:
-            for i in vlan_ports_list[0]['permit_vlanid']:
+    for item in vlan_ports_list:
+        for i in vlan_ports_list[0]['permit_vlanid']:
+            try:
                 duthost.command('ip route flush {}'.format(
                     item['permit_vlanid'][i]['remote_ip']))
+            except RunAnsibleModuleFail as e:
+                logger.error(e)
 
-        for vlan_port in vlan_ports_list:
-            for permit_vlanid in vlan_port["permit_vlanid"].keys():
-                if int(permit_vlanid) != vlan_port["pvid"]:
+    for vlan_port in vlan_ports_list:
+        for permit_vlanid in vlan_port["permit_vlanid"].keys():
+            if int(permit_vlanid) != vlan_port["pvid"]:
+                try:
                     ptfhost.command("ip link delete eth{idx}.{pvid}".format(
                     idx=vlan_port["port_index"][0],
                     pvid=permit_vlanid
                     ))
-    except RunAnsibleModuleFail as e:
-        logger.error(e)
+                except RunAnsibleModuleFail as e:
+                    logger.error(e)
 
     duthost.shell("config reload -y &>/dev/null", executable="/bin/bash")
```

Comment:
If config reload is performed anyway, then it's unnecessary to run `ip route flush` on duthost.

Reply:
Updated this now.